ocaml/xapi/storage_mux.ml

```diff
@@ -151,6 +153,49 @@ module Mux = struct
       )
   end
 
+  module DP_info = struct
+    type t = {sr: Sr.t; vdi: Vdi.t; vm: Vm.t} [@@deriving rpcty]
+
+    let storage_dp_path = "/var/run/nonpersistent/xapi/storage-dps"
```

Comment:
What happens if you do a toolstack-restart on a running system that had the old datapaths, will the new XAPI be able to find them, or will we require a host reboot?

Reply:
We definitely require a host evacuate, and possibly a reboot. Nothing we can do to avoid unfortunately, the implementation of the original SMAPIv3 adapter violated its design and architectural constraints meaning that it is not possible to make the Storage Management scripts work as the design and architecture says they should as they do not currently get called correctly.